irohad/consensus/yac/impl/yac.cpp

```diff
@@ -150,13 +172,17 @@ namespace iroha {
 
         network_->sendState(current_leader, {vote});
         cluster_order_.switchToNext();
-        if (cluster_order_.hasNext()) {
+        auto has_next = cluster_order_.hasNext();
+        lock.unlock();
+        if (has_next) {
           timer_->invokeAfterDelay([this, vote] { this->votingStep(vote); });
         }
       }
 
-      void Yac::closeRound() {
-        timer_->deny();
+      void Yac::closeRound(Round round) {
```

Comment:
as for me, the `closeRound` function should better take no `round` argument, because we do not maintain more than one round at a time. Better would be to call it only if the `proposal_round` matches the condition here:
https://github.com/hyperledger/iroha/blob/6b97c3e467ca628d3a1e7f44a22b7d3420498813/irohad/consensus/yac/impl/yac.cpp#L247
```c++
                  if (proposal_round >= round_) {
                    this->closeRound();
                  }
```